File: frontend/src/models/Notification.ts
import {ClientBasicInfo} from "./Client";
import {SpecialistBasicInfo} from "./Specialist";
import {VisitBasicInfo} from "./Visit";

export interface Notification {
    id: number
    type: string
    notifier: string
    read: boolean
    created_at: Date
    client: ClientBasicInfo
    specialist: SpecialistBasicInfo
    visit: VisitBasicInfo
}

export interface NotificationRequest {
    type: string
    notifier: string
    client_id: number
    specialist_id: number
    visit_id: number
}

File: backend/internal/models/Notification.go
package models

import "time"

type Notification struct {
	Id         int                 `json:"id"`
	Type       string              `json:"type"`
	Notifier   string              `json:"notifier"`
	Read       bool                `json:"read"`
	CreatedAt  time.Time           `json:"created_at"`
	Client     ClientBasicInfo     `json:"client"`
	Specialist SpecialistBasicInfo `json:"specialist"`
	Visit      VisitBasicInfo      `json:"visit"`
}

type NotificationRequest struct {
	Type         string `json:"type"`
	Notifier     string `json:"notifier"`
	ClientId     int    `json:"client_id"`
	SpecialistId int    `json:"specialist_id"`
	VisitId      int    `json:"visit_id"`
}
